native/src/main/java/io/ballerina/stdlib/http/api/HttpSwaggerUiResource.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.api;
+
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.util.CharsetUtil;
+
+/**
+ * {@code HttpSwaggerUiResource} is the resource which respond the Swagger-UI for the generated Open API specification.
+ *
+ * @since v2.11.2
+ */
+public class HttpSwaggerUiResource extends HttpOASResource {
+
+    private static final String STATIC_HTML_PAGE = """
+                    <!DOCTYPE html>
+                    <html lang="en">
+                    <head>
+                      <meta charset="utf-8" />
+                      <meta name="viewport" content="width=device-width, initial-scale=1" />
+                      <meta name="description" content="SwaggerUI" />
+                      <title>SwaggerUI</title>
+                      <link rel="stylesheet" href="https://unpkg.com/swagger-ui-dist@5.11.0/swagger-ui.css" />
+                    </head>
+                    <body>
+                    <div id="swagger-ui"></div>
+                    <script src="https://unpkg.com/swagger-ui-dist@5.11.0/swagger-ui-bundle.js" crossorigin></script>
+                    <script>
+                      window.onload = () => {
+                        window.ui = SwaggerUIBundle({
```

Comment:
If the OpenAPI spec has errors, will it be shown in the HTML. If so can you attach a sample screenshot for that as well.

Reply:
Updated the PR with a screenshot of the swagger-ui generated for a service which has an erroneous OpenAPI spec.

Reply:
Adding to the above; to generate/attach an invalid OpenAPI spec, user has to manually update the HTTP service like below:
```ballerina
@http:ServiceConfig {
    openApiDefinition: invalidOpenApiSpec.toJsonString().toBytes()
}
service /hello on new http:Listener(9091) {
    // ...
}
```

But this is not a common case as the intended behaviour is to generate the spec automatically from the OpenAPI built-in plugin.

Reply:
Yes, agreed. I just wanted to check how it will show the errors if the automatically generated OpenAPI has some errors